### PR TITLE
Refactor System Tests

### PR DESCRIPTION
**Pull request by orduno**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | # |
| Primary OS tested on | Ubuntu 18.04 |
| Robotic platform tested on | Gazebo simulation of TB3 |

---

## Description of contribution in a few bullet points

Adding multi-robot system tests and addressing #1135 and #813.

* Created a `NavTester` node which can be used for testing with single or multiple robots.
* Added launch files to test multiple robots.
* Added TB3 model files needed for running tests on ROS build farm.
* Updated `spawn_robot` to accept any robot sdf description.

### CODE REVIEW

**Comment by SteveMacenski** [comment]
I'm deploying an ament_download that's equal to catkin_download for you

samsung-ros/ament_download#1

**Comment by orduno** [comment]
Yes, I'd prefer that option. However, I know we recently modified some TB files to work with the addition of `--ros-args`. I'll give it a try.

**Comment by orduno** [comment]
Actually, can we leave that change to a follow-up PR? I'm already doing a bit of changes.

**Comment by SteveMacenski** [comment]
that's fine

File: nav2_bt_navigator/behavior_trees/navigate_w_replanning_and_recovery.xml [hunk 1]
```diff
@@ -1,13 +1,9 @@
-<!--
-  This Behavior Tree replans the global path periodically at 1 Hz and it also has
-  recovery actions.
--->
 <root main_tree_to_execute="MainTree">
   <BehaviorTree ID="MainTree">
-    <RecoveryNode number_of_retries="6" name="NavigateRecovery">
+    <RecoveryNode number_of_retries="6">
```
Comment (crdelsey) [inline nav2_bt_navigator/behavior_trees/navigate_w_replanning_and_recovery.xml]: Why are we removing the names here?

Comment (orduno) [reply nav2_bt_navigator/behavior_trees/navigate_w_replanning_and_recovery.xml]: hmm, I'll add the names back. Not sure why these got deleted.

**Pull request merged**